LBH.AdultSocialCare.Api/V1/UseCase/Payments/Concrete/CreateDraftPayRunUseCase.cs

```diff
@@ -28,29 +27,48 @@ public CreateDraftPayRunUseCase(IPayRunGateway payRunGateway, IQueueService payr
 
         public async Task CreateDraftPayRun(DraftPayRunCreationDomain draftPayRunCreationDomain)
         {
-            var draftPayRunCount = await _payRunGateway.GetDraftPayRunCount(draftPayRunCreationDomain.Type);
-            if (draftPayRunCount > 0)
-                throw new ApiException($"A Pay Run with draft status for {draftPayRunCreationDomain.Type.ToDescription()} already exists!");
-
+            var recurringPayRunTypes = new[] { PayrunType.DirectPayments, PayrunType.ResidentialRecurring };
             ValidateDraftPayRun(draftPayRunCreationDomain);
 
+            var unApprovedPayRunExists = await _payRunGateway.CheckExistsUnApprovedPayRunAsync(draftPayRunCreationDomain.Type);
+            if (unApprovedPayRunExists)
+                throw new ApiException($"Operation not allowed. There exists a pay run of the same type that is not approved", HttpStatusCode.PreconditionFailed);
+
             draftPayRunCreationDomain.Status = PayrunStatus.Draft;
-            draftPayRunCreationDomain.StartDate = await _payRunGateway.GetDateOfLastPayRun(draftPayRunCreationDomain.Type);
+
+            if (recurringPayRunTypes.Contains(draftPayRunCreationDomain.Type))
+            {
+                draftPayRunCreationDomain.StartDate = await _payRunGateway.GetEndDateOfLastPayRun(draftPayRunCreationDomain.Type);
```

Comment:
StartDate should start +1 day

await _payRunGateway.GetEndDateOfLastPayRun(draftPayRunCreationDomain.Type).AddDay(1);

Reply:
Done